custom_components/innova/coordinator.py

```diff
@@ -15,15 +15,19 @@ def __init__(
         name: str,
         update_interval: timedelta,
     ):
+        """Initialize the Innova Coordinator."""
         self.innova = innova
 
+        # Fetch the user-configurable update_interval in seconds from Home Assistant's data, or default to 300 seconds (5 minutes)
+        interval_in_seconds = hass.data.get("innova_update_interval_seconds", 300)
```

Comment:
I will need to test the code to be sure, but I don't think we need to re-fetch the value here as we just received it as a construction parameter.

Reply:
Good catch, I missed that and I've removed it.